pulser-core/pulser/sequence/_seq_drawer.py

```diff
@@ -248,133 +244,63 @@ def gather_data(seq: pulser.sequence.Sequence, gather_output: bool) -> dict:
-        """Formats a phase value for printing."""
-        value = (((phi + np.pi) % (2 * np.pi)) - np.pi) / np.pi
-        if value == -1:
-            return r"$\pi$"
-        elif value == 0:
-            return "0"  # pragma: no cover - just for safety
-        else:
-            return rf"{value:.2g}$\pi$"
-
-    n_channels = len(seq.declared_channels)
+    n_channels = len(seqsamples.channels)
     if not n_channels:
         raise RuntimeError("Can't draw an empty sequence.")
-    data = gather_data(seq, gather_output=draw_modulation)
+
+    data = gather_data(seqsamples, gather_output=draw_modulation)
     total_duration = data["total_duration"]
+
     time_scale = 1e3 if total_duration > 1e4 else 1
-    for ch in seq._schedule:
+    for ch in seqsamples.channels:
         if np.count_nonzero(data[ch].samples.det) > 0:
             data[ch].curves_on["detuning"] = True
         if draw_phase_curve and np.count_nonzero(data[ch].samples.phase) > 0:
             data[ch].curves_on["phase"] = True
 
     # Boxes for qubit and phase text
     q_box = dict(boxstyle="round", facecolor="orange")
-    ph_box = dict(boxstyle="round", facecolor="ghostwhite")
-    area_ph_box = dict(boxstyle="round", facecolor="ghostwhite", alpha=0.7)
-    slm_box = dict(boxstyle="round", alpha=0.4, facecolor="grey", hatch="//")
     eom_box = dict(boxstyle="round", facecolor="lightsteelblue")
+    slm_box = dict(boxstyle="round", alpha=0.4, facecolor="grey", hatch="//")
 
-    # Draw masked register
-    if draw_register:
-        pos = np.array(seq.register._coords)
-        if isinstance(seq.register, Register3D):
-            labels = "xyz"
-            fig_reg, axes_reg = seq.register._initialize_fig_axes_projection(
-                pos,
-                blockade_radius=35,
-                draw_half_radius=True,
-            )
-            fig_reg.get_layout_engine().set(w_pad=6.5)
-
-            for ax_reg, (ix, iy) in zip(
-                axes_reg, combinations(np.arange(3), 2)
-            ):
-                seq.register._draw_2D(
-                    ax=ax_reg,
-                    pos=pos,
-                    ids=seq.register._ids,
-                    plane=(ix, iy),
-                    masked_qubits=seq._slm_mask_targets,
-                )
-                ax_reg.set_title(
-                    "Masked register projected onto\n the "
-                    + labels[ix]
-                    + labels[iy]
-                    + "-plane"
-                )
-
-        elif isinstance(seq.register, Register):
-            fig_reg, ax_reg = seq.register._initialize_fig_axes(
-                pos,
-                blockade_radius=35,
-                draw_half_radius=True,
-            )
-            seq.register._draw_2D(
-                ax=ax_reg,
-                pos=pos,
-                ids=seq.register._ids,
-                masked_qubits=seq._slm_mask_targets,
-            )
-            ax_reg.set_title("Masked register", pad=10)
```

Comment:
Yes I think so, see my comment above in the declaration of `draw_samples`, I think you should replace `draw_register` by an optional `BaseRegister`: if it is provided then you show the register, otherwise you don't. In `draw_sequence`, you will then simply call `draw_samples` and provide the `Sequence.register` to it.